framework/PostProcessors/ParetoFrontierPostProcessor.py

```diff
@@ -0,0 +1,147 @@
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(ParetoFrontier, cls).getInputSpecification()
+    inputSpecification.addSub(InputData.parameterInputFactory('costID' , contentType=InputTypes.StringType))
+    inputSpecification.addSub(InputData.parameterInputFactory('valueID', contentType=InputTypes.StringType))
+    return inputSpecification
+
+  def initialize(self, runInfo, inputs, initDict) :
+    """
+      Method to initialize the PostProcessor
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+      @ Out, None
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+    paramInput = ParetoFrontier.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+    self._handleInput(paramInput)
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the parsed paramInput for this class.
+      @ In, paramInput, ParameterInput, the already-parsed input.
+      @ Out, None
+    """
+    costID  = paramInput.findFirst('costID')
+    self.costID  = costID.value
+    valueID = paramInput.findFirst('valueID')
+    self.valueID = valueID.value
+
+  def inputToInternal(self, currentInp):
+    """
+      Method to convert an input object into the internal format that is
+      understandable by this pp.
+      In this case, we only want data objects!
+      @ In, currentInp, list, an object that needs to be converted
+      @ Out, currentInp, DataObject.HistorySet, input data
+    """
+    if len(currentInp) > 1:
+      self.raiseAnError(IOError, 'ParetoFrontier postprocessor {} expects one input DataObject, but received {} inputs!".'
+                                  .format(self.name,len(currentInp)))
+    currentInp = currentInp[0]
+    if currentInp.type not in ['PointSet']:
+      self.raiseAnError(IOError, 'ParetoFrontier postprocessor "{}" requires a DataObject input! Got "{}".'
+                                 .format(self.name, currentInput.type))
+    return currentInp
+
+  def run(self, inputIn):
+    """
+      This method executes the postprocessor action.
+      @ In, inputIn, object, object contained the data to process. (inputToInternal output)
+      @ Out, realizations, list, list of realizations obtained
```

Comment:
inputIn, are data object (point set)
output realizations are dict not list.

Reply:
edited